open_gpt/models/generation.py

```diff
@@ -102,11 +125,14 @@ def step_generate(
             input_ids = input_ids[:, -max_src_len:]
             input_length = max_src_len
 
-        past_key_values = next_token = None
+        next_token = None
+        past_key_values = past_key_values if past_key_values is not None else None
 
         for step in range(max_new_tokens):
             if step == 0:
-                outputs = self.model(input_ids, use_cache=True)
+                outputs = self.model(
+                    input_ids, use_cache=True, past_key_values=past_key_values
+                )
```

Comment:
In my view, this is where cache happens. If the hash of `input_ids` hits the cache, we can directly get the outputs from the cache. Otherwise, we use `model.forward` to compute the context info, and push them into our cache.

Reply:
BTW, please double check what's the meaning of `use_cache=True` here? @jemmyshin